sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -84,43 +84,24 @@ namespace Azure { namespace Core { namespace Http {
   /**
    * HTTP request method.
    */
-  enum class HttpMethod
-  {
-    Get, ///< GET
-    Head, ///< HEAD
-    Post, ///< POST
-    Put, ///< PUT
-    Delete, ///< DELETE
-    Patch, ///< PATCH
-  };
+  class HttpMethod {
+  public:
+    HttpMethod() = default;
```

Comment:
why do we want to keep this. It will give us an empty string for the HttpMethod. Consider changing to `HttpMethod() = delete;`

Reply:
Following the other extensible enums and the guidelines
I was following the other extensible enums in the SDK. Should we make similar changes to all of them (and update the guidelines)?

For example:
https://github.com/Azure/azure-sdk-for-cpp/blob/63b2d21bc65a4348cad5fd69cecbe353f74b7120/sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp#L60-L74

https://github.com/Azure/azure-sdk-for-cpp/search?l=C%2B%2B&q=extensible
https://azure.github.io/azure-sdk/cpp_implementation.html#enumerations

Reply:
@CaseyCarter should we update the guidelines of extensible enum to delete the default ctor, or would that contradict some other guidance? @vhvb1989 brought up a good point that we probably don't want to create value-less instances.

@Jinming-Hu  and @katmsft are there scenarios where the storage extensible enums need to be constructed without a value?

Reply:
OK, looks like the default ctor is being used in `blob_rest_client` a lot, for example, this model uses the default ctor of the `CopyStatus` extensible enum:
https://github.com/Azure/azure-sdk-for-cpp/blob/93bc8c42185e781f3e40fb10aa77bde92ac317a2/sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp#L8435

Reply:
If an extensible enum is not default-constructible, any struct that has this enum as member won't be default-constructible. We have a lot of structs that have to be default-constructible, so we decide to make every extensible enum default-contructible.

You don't have to do the same as we do in storage sdk.